File: app/src/main/java/com/gamalinda/android/poc/testing/collapsingtoolbar/view/TextItemViewHolder.java
package com.gamalinda.android.poc.testing.collapsingtoolbar.view;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.gamalinda.android.poc.testing.collapsingtoolbar.R;

public class TextItemViewHolder extends RecyclerView.ViewHolder {
    private TextView textView;
    private View separator;

    public TextItemViewHolder(View itemView) {
        super(itemView);
        textView = itemView.findViewById(R.id.list_item);
        separator = itemView.findViewById(R.id.separator);
    }

    public void bind(String text) {
        textView.setText(text);
    }

    public void hideSeparator(boolean hide) {
        if (hide) {
            separator.setVisibility(View.GONE);
        } else {
            separator.setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/gamalinda/android/poc/testing/collapsingtoolbar/view/TextItemViewHolder.kt
package com.gamalinda.android.poc.testing.collapsingtoolbar.view

import android.view.View
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import com.gamalinda.android.poc.testing.collapsingtoolbar.R

class TextItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val textView: TextView = itemView.findViewById(R.id.list_item)
    private val separator: View = itemView.findViewById(R.id.separator)

    fun bind(text: String) {
        textView.text = text
    }

    fun hideSeparator(hide: Boolean) {
        if (hide) {
            separator.visibility = View.GONE
        } else {
            separator.visibility = View.VISIBLE
        }
    }
}
